_posts/2017/2017-16-01-testing-async-interactions.md

```diff
@@ -0,0 +1,246 @@
+---
+layout: post
+title: "Testing asynchronous interactions in distributed systems"
+date: "2017-01-15 12:11:00"
+image: blog-header/testing-async-interactions.jpg
+categories: tech-stories
+authors: ["Mathias"]
+---
+
+In a distributed architecture, we strive to build self-contained services.
+We try hard to make these services as autonomous as possible, but it is often impossible to build a service that can fulfill its tasks without interaction with other services.
+
+When it comes to service interaction we try to use an event-based approach as much as possible.
+This means interacting asynchronously and outside the user interaction where this is applicable.
+
+Going this route you will soon ask yourself how to test such asynchronous interactions.
+[Spring Cloud Contract](https://cloud.spring.io/spring-cloud-contract/) offers great support for such scenarios.
+
+Let's look at the problem using a simple application consisting of two services and see how Spring Cloud Contract can help us to improve testing the asynchronous interaction between them.
```

Comment:
Let's address this challenge using a simple application consisting of two services and see how Spring Cloud Contract can help improve testing the asynchronous interaction between them.